Sources/OpenCombine/Published.swift

```diff
@@ -87,13 +87,27 @@ public struct Published<Value> {
             return object[keyPath: storageKeyPath].value
         }
         set {
+            if object[keyPath: storageKeyPath].objectWillChange == nil {
+                object[keyPath: storageKeyPath].objectWillChange = getPublisher(object)
+            }
             object[keyPath: storageKeyPath].objectWillChange?.send()
             object[keyPath: storageKeyPath].publisher?.subject.send(newValue)
             object[keyPath: storageKeyPath].value = newValue
         }
         // TODO: Benchmark and explore a possibility to use _modify
     }
 }
+
+private func getPublisher<Observable:ObservableObject>(_ observable: Observable)
+                                                       -> ObservableObjectPublisher? {
+    return observable.objectWillChange as? ObservableObjectPublisher
+}
+
+private func getPublisher<NonObservable>(_ observable: NonObservable)
+                                         -> ObservableObjectPublisher? {
+    return nil
+}
```

Comment:
I don't think this will work. Function overloads are resolved at compile time, not at run time. And because `object` in the setter is not constrained by `ObservableObject`, the first `getPublisher` will never ever be called.

Reply:
oh damn, true

Reply:
I guess there is really no way to make it work then?

Reply:
Only something like #97 :(